app/models/ext_management_system.rb

```diff
@@ -402,7 +402,13 @@ def emstype_description
   def with_provider_connection(options = {})
     raise _("no block given") unless block_given?
     _log.info("Connecting through #{self.class.name}: [#{name}]")
-    yield connect(options)
+    connection = connect(options)
+    yield connection
+  ensure
+    disconnect(connection) if connection
```

Comment:
@borod108 I want to standardize on an ensure disconnect in core, do you need the rescue around your disconnect method?  https://github.com/ManageIQ/manageiq-providers-ovirt/blob/master/app/models/manageiq/providers/redhat/infra_manager/api_integration.rb#L113
Looks like that rescue might be better in the disconnect method itself.